src/main/kotlin/app/revanced/patches/reddit/testing/patch/TotalAdPostPatch.kt

```diff
@@ -0,0 +1,179 @@
+                new-instance v1, Ljava/util/ArrayList;
+                invoke-direct {v1}, Ljava/util/ArrayList;-><init>()V
+                check-cast v1, Ljava/util/Collection;
+                
+                # v3 = p1.iterator()
+                invoke-interface {p1}, Ljava/lang/Iterable;->iterator()Ljava/util/Iterator;
+                move-result-object v3
+                :continue
+            
+                # Check if the iterator has next
+                invoke-interface {v3}, Ljava/util/Iterator;->hasNext()Z
+                move-result v2
+            
+                # Exit loop if no next
+                if-eqz v2, :exit_loop
+                invoke-interface {v3}, Ljava/util/Iterator;->next()Ljava/lang/Object;
+
+                # Check if the object is an instance of ILink
+                move-result-object v2
+                instance-of v0, v2, Lcom/reddit/domain/model/ILink;
+                if-eqz v0, :afterLink
+
+                # ---------------------------------------
+                # If it's an ILink object, check if it's promoted
+                check-cast v2, Lcom/reddit/domain/model/ILink;
+                invoke-virtual {v2}, Lcom/reddit/domain/model/ILink;->getPromoted()Z
+                move-result v0
+            
+                # Continue loop if it's promoted
+                if-nez v0, :continue
+                # ---------------------------------------
+                :afterLink
+                
+                # Add the object to the filtered list
+                invoke-interface {v1, v2}, Ljava/util/Collection;->add(Ljava/lang/Object;)Z
+                goto :continue
+            
+                :exit_loop
+                # Return the filtered list
+                return-object v1
+                """
+            )
+        }
+        result.mutableClass.methods.add(filteringMethod)
+    }
+
+    /**
+     * Applies an older patch which removes almost all promoted posts.
+     * Does not work on Popular and Latest.
+     *
+     * Almost all posts are represented in the reddit code with the Link class.
+     * The Link class is wrapped in a Listing, which contains a list of posts.
+     * By filtering the listing it becomes possible to remove any promoted posts.
+     **/
+    private fun applyOldPatch() {
+        val result = AdPostFingerprint.result!!
+        val method = result.mutableMethod
+        addFilterChildrenMethod(result)
+
+        // Looks for an iput-object with the variable name children:
+        // The children attribute is a list of posts, which need filtering.
+        // iput-object p1, p0, Lcom/reddit/domain/model/listing/Listing;->children:Ljava/util/List;
+        val instruction = method.implementation!!.instructions.withIndex().first {
+            it.value.opcode == Opcode.IPUT_OBJECT
+                    && (((it.value as? Instruction22c)?.reference as? DexBackedFieldReference)?.name == "children")
```

Comment:
you can do `(it as? ReferenceInstruction)?.reference as? FieldReference)?.name)`